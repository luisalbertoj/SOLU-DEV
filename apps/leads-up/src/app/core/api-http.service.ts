import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
