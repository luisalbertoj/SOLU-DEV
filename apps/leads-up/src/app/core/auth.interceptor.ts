import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
