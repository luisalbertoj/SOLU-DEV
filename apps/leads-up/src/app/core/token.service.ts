import { Injectable } from '@angular/core';

@Injectable()
export class TokenProvider {
  load() {
    return new Promise((resolve) => {
      resolve(true);
    });
  }
}
