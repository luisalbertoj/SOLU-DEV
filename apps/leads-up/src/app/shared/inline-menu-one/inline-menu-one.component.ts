import {
  animate,
  state,
  style,
  transition,
  trigger,
} from '@angular/animations';
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterModule } from '@angular/router';
import { DashboardMenuOneComponent } from '../dashboard-menu-one/dashboard-menu-one.component';

@Component({
  selector: 'solu-dev-inline-menu-one',
  standalone: true,
  imports: [CommonModule, RouterModule],
  templateUrl: './inline-menu-one.component.html',
  styleUrls: ['./inline-menu-one.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  animations: [
    trigger('inline', [
      state(
        'hidden',
        style({
          height: '0px',
          overflow: 'hidden',
        })
      ),
      state(
        'visible',
        style({
          height: '*',
        })
      ),
      state(
        'hiddenAnimated',
        style({
          height: '0px',
          overflow: 'hidden',
        })
      ),
      state(
        'visibleAnimated',
        style({
          height: '*',
        })
      ),
      transition(
        'visibleAnimated => hiddenAnimated',
        animate('400ms cubic-bezier(0.86, 0, 0.07, 1)')
      ),
      transition(
        'hiddenAnimated => visibleAnimated',
        animate('400ms cubic-bezier(0.86, 0, 0.07, 1)')
      ),
    ]),
  ],
})
export class InlineMenuOneComponent {
  constructor(public appMain: DashboardMenuOneComponent) {}
}
