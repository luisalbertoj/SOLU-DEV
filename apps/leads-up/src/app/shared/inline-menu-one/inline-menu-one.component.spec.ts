import { ComponentFixture, TestBed } from '@angular/core/testing';

import { InlineMenuOneComponent } from './inline-menu-one.component';

describe('InlineMenuOneComponent', () => {
  let component: InlineMenuOneComponent;
  let fixture: ComponentFixture<InlineMenuOneComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [InlineMenuOneComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(InlineMenuOneComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
