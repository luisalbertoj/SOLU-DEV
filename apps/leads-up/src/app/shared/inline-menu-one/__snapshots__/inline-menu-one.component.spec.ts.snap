// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineMenuOneComponent this should be equal to the snapshot 1`] = `
"<div class=\\"ng-tns-c61-0 layout-inline-menu ng-star-inserted\\" ng-reflect-ng-class=\\"[object Object]\\"><a class=\\"layout-inline-menu-action ng-tns-c61-0\\"><img src=\\"assets/layout/images/profile-image.png\\" alt=\\"avatar\\" style=\\"width: 44px; height: 44px;\\" class=\\"ng-tns-c61-0\\"><span class=\\"layout-inline-menu-text ng-tns-c61-0\\">User Test</span><i class=\\"layout-inline-menu-icon pi pi-angle-down ng-tns-c61-0\\"></i></a><ul class=\\"layout-inline-menu-action-panel ng-tns-c61-0 ng-trigger ng-trigger-inline\\" style=\\"height:0px;overflow:hidden;0:height;1:overflow;\\"><li class=\\"layout-inline-menu-action-item ng-tns-c61-0\\"><a routerlinkactive=\\"router-link-active\\" class=\\"ng-tns-c61-0\\" ng-reflect-router-link-active=\\"router-link-active\\" ng-reflect-router-link=\\"/logout\\" href=\\"/logout\\"><i class=\\"pi pi-power-off pi-fw ng-tns-c61-0\\"></i><span class=\\"ng-tns-c61-0\\">Logout</span></a></li><li class=\\"layout-inline-menu-action-item ng-tns-c61-0\\"><a class=\\"ng-tns-c61-0\\"><i class=\\"pi pi-cog pi-fw ng-tns-c61-0\\"></i><span class=\\"ng-tns-c61-0\\">Settings</span></a></li><li class=\\"layout-inline-menu-action-item ng-tns-c61-0\\"><a class=\\"ng-tns-c61-0\\"><i class=\\"pi pi-user pi-fw ng-tns-c61-0\\"></i><span class=\\"ng-tns-c61-0\\">Profile</span></a></li></ul></div><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}-->"
`;
